glean-core/ios/Glean/Dispatchers.swift

```diff
@@ -88,17 +82,72 @@ class Dispatchers {
         } else {
             // If we are not queuing initial tasks, we can go ahead and execute the Operation by
             // adding it to the `operationQueue`
+            serialOperationQueue.addOperation(block)
+
+            // If we are in testing mode, go ahead and wait until it is finished before continuing
+            // to ensure synchronous execution of the task.
             if testingMode {
-                // If we are in testing mode, go ahead and execute the operation and wait until it
-                // is finished before continuing to give us synchronous execution of the task.
-                operation.start()
-                operation.waitUntilFinished()
-            } else {
-                operationQueue.addOperation(operation)
+                serialOperationQueue.waitUntilAllOperationsAreFinished()
             }
         }
+    }
+
+    /// This function launches a background `Operation` on a concurrently executed queue.
+    ///
+    /// - parameters:
+    ///   * operation: The `Operation` to execute
+    ///
+    /// This function is used to execute tasks in an asynchrounous manner and still give us the ability
+    /// to cancel the tasks by creating them as `Operation`s rather than using GCD.
+    ///
+    /// This function specifically ignores the `queueInitialTasks` flag because the only tasks that
+    /// should be launched by this are the ping upload schedulers and those should run regardless of
+    /// the initialized state.
+    ///
+    /// If `testingMode` is enabled, then launch will just execute the task rather than adding it to the
+    /// concurrent queue to avoid asynchronous issues while testing
+    func launchConcurrent(operation: Operation) {
+        if testingMode {
+            operation.start()
+            operation.waitUntilFinished()
+        } else {
+            concurrentOperationsQueue.addOperation(operation)
+        }
+    }
 
-        return operation
+    /// This function launches a background block of code on a concurrently executed queue.
```

Comment:
The blcok isn't "background". IMO leaving it out is just as clear.
```suggestion
    /// This function launches a block of code on a concurrently executed queue.
```